migration/secure_note_transformer.py

```diff
@@ -0,0 +1,206 @@
+import logging
+import subprocess
+import json
+from dataclasses import dataclass
+
+import utils
+
+
+@dataclass
+class LPassData:
+    url: str
+    username: str
+    password: str
+    otp_secret: str
+    notes: str
+    title: str
+    vault: str
+
+
+class SecureNoteTransformer:
+    def __init__(self, lpass_raw_data: LPassData):
+        self.lpass_raw_data = lpass_raw_data
+
+    def transform(self):
+        template = None
+        parsed_data = self._parse()
+        if not parsed_data:
+            return
+
+        # Create Secure Note template
+        if isinstance(parsed_data, LPassData):
+            template = fetch_template("Secure Note")
+            self._map_secure_note(parsed_data, template)
+            return template
+
+        note_type = parsed_data["NoteType"]
+        if note_type == "Credit Card":
+            template = fetch_template("Credit Card")
+            self._map_credit_card(parsed_data, template)
+            return template
+        elif note_type == "Bank Account":
+            template = fetch_template("Bank Account")
+            self._map_bank_account(parsed_data, template)
```

Comment:
Why do we need to get the template?
From the looks of it, the mapping function adds all the details that are part of the template itself except for the category of the item (which is used to fetch the template). Would it help to simplify this by adding the category in the template itself when the mapping function is called?

Reply:
This might be a hangover from an earlier version in which the template from `op item template get login` was compared to a hard-coded version of the template and would fail with an error if they mis-matched (to prevent an import from proceeding that created a bunch of incorrect items if our template ever changed. 

I don't know if this check is strictly necessary since I _suspect_ any major changes to our templates that broke any mapping would result in an error anyway.

Reply:
I removed the check with comparison for the fetched and hard coded templates. It will fetch a template anyway.
The only thing I'd like to add here is a cashing layer. If we fetch some template, store it and reuse it, do not fetch it again.
I'll add it in the following PR.